sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -950,11 +955,10 @@ def expand(self, pcoll):
           retry_strategy=self.insert_retry_strategy,
           test_client=self.test_client)
 
-      # TODO: Use utility functions from BQTools
-      table_fn = self._get_table_fn()
-
       outputs = (pcoll
-                 | 'AppendDestination' >> beam.Map(lambda x: (table_fn(x), x))
+                 | 'AppendDestination' >> beam.ParDo(
+                     bigquery_tools.AppendDestinationsFn(
```

Comment:
Does this also fix dynamic destinations ? Please update the title and description of the PR accordingly.

Reply:
Dynamic destinations was already supported. This only makes it use the bq_tools utility instead of repeating the code. Thanks!